runelite-client/src/main/java/net/runelite/client/plugins/runenergy/RunEnergyPlugin.java

```diff
@@ -169,7 +170,13 @@ public void onGameTick(GameTick event)
 			prevLocalPlayerLocation.distanceTo(client.getLocalPlayer().getWorldLocation()) > 1;
 
 		prevLocalPlayerLocation = client.getLocalPlayer().getWorldLocation();
+	}
 
+	@Subscribe
+	public void onBeforeRender(BeforeRender beforeRender)
+	{
+		// This text is also set by the client script for updating the orbs,
+		// set the text before every frame is drawn to prevent flicker.
 		if (energyConfig.replaceOrbText())
 		{
 			setRunOrbText(getEstimatedRunTimeRemaining(true));
```

Comment:
This should probably precompute the `getEstimatedRunTimeRemaining` and just set that each frame